src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js

```diff
@@ -125,10 +125,12 @@ class BasePaymentsPage extends React.Component {
      * @param {Boolean} isDefault
      */
     paymentMethodPressed(nativeEvent, accountType, account, isDefault) {
-        const position = getClickedElementLocation(nativeEvent);
+        const position = getClickedTargetLocation(nativeEvent.currentTarget);
+
         this.setState({
-            addPaymentMethodButton: nativeEvent,
+            addPaymentMethodButton: nativeEvent.currentTarget,
```

Comment:
I know this was not added in this PR, but why are we storing a reference to an HTML element in React component state? This seems like a strange practice.

Reply:
I think we can move this HTML element to a class property. Thoughts? @mananjadhav @marcaaron

Reply:
> I am OK with where this landed, but I think @mananjadhav's first instinct was correct here. Better to focus on the bug fix first and refactoring later.

@marcaaron may I know what are you trying to propose please?

Reply:
That seems better to me as long as we are certain that we are not dependent on the call to `setState()` (e.g. to trigger a re-render for some reason).

Reply:
I think moving to the class property won't work when we resize the window.

Reply:
> > I am OK with where this landed, but I think @mananjadhav's first instinct was correct here. Better to focus on the bug fix first and refactoring later.
> > 
> @marcaaron may I know what are you trying to propose please?

There's not action for this statement. What @marcaaron meant is before we refactor any piece of code (like we renamed the function), we should first focus on fixing the bug. Once we're confident all is working as expected, we can then refactor/enhance the readability of the code. Hope it makes sense :)

Reply:
>I think moving to the class property won't work when we resize the window. Because we store the element the bounding rectangle gives us the correct coordinates.

Hmm I may have to take a closer look here but my thinking is that when the window is resized that we would have to remeasure the height, but unsure why having the element reference in state makes any difference.

Reply:
There's some justification [here](https://github.com/Expensify/App/pull/7847#discussion_r816907027) - but it really doesn't make sense to me 🤷‍♂️ 

Anyway, wasn't introduced in this PR so I am fine if we want to do nothing and create a new issue to improve the way this works.